contrib/generic_proxy/filters/network/source/upstream.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/network/connection.h"
+#include "envoy/upstream/thread_local_cluster.h"
+
+#include "contrib/generic_proxy/filters/network/source/interface/codec.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace GenericProxy {
+
+class UpstreamManagerImplBase : public Envoy::Event::DeferredDeletable,
```

Comment:
Maybe we can call it as `UpstreamConnection`?

Reply:
sgtm.